File: 1686-stone-game-vi/1686-stone-game-vi.cpp
class Solution {
public:
    
    int stoneGameVI(vector<int>& aliceValues, vector<int>& bobValues) {
        
        vector<vector<int>>vect;
        for(int i=0;i<aliceValues.size();i++){
            vect.push_back({-aliceValues[i]-bobValues[i],aliceValues[i],bobValues[i]});
        }
        sort(vect.begin(),vect.end());
        int alice=0,bob=0;
        for(int i=0;i<aliceValues.size();i++)
        {
            if(i%2==0){
                alice+=vect[i][1];
            }
            else bob+=vect[i][2];
        }
        if(alice>bob)return 1;
        else if(alice==bob)return 0;
        return -1;
    }
};

File: 1686-stone-game-vi/1686-stone-game-vi.java
class Solution {
    public int stoneGameVI(int[] aliceValues, int[] bobValues) {
        
        int n=aliceValues.length;
        int [][]arr=new int[n][];
        for(int i=0;i<n;i++){
            arr[i]=new int []{aliceValues[i]+bobValues[i],aliceValues[i],bobValues[i]};
        }
        Arrays.sort(arr,(a,b)->Integer.compare(b[0],a[0]));
        int a=0,b=0;
        for(int i=0;i<n;i++){
            if(i%2==0){
                a+=arr[i][1];
            }
            else b+=arr[i][2];
        }
        if(a>b)return 1;
        else if(a==b)return 0;
        return -1;
    }
}